tensorflow/python/eager/def_function.py

```diff
@@ -910,14 +910,16 @@ def fn_with_cond(*inner_args, **inner_kwds):
           condition,
           lambda: self._stateless_fn(*inner_args, **inner_kwds),
           functools.partial(self._concrete_stateful_fn._filtered_call,  # pylint: disable=protected-access
-                            inner_args, inner_kwds))
+                            inner_flat_args, inner_flat_kwds))
 
     # We've created variables and are unable to lift the initialization graphs,
     # so we fall back to initializing with conds while running the function.
-    canon_args, canon_kwds = \
+    canon_args, canon_kwds, flat_args, flat_kwds = \
         self._stateful_fn._function_spec.canonicalize_function_inputs(  # pylint: disable=protected-access
             *args, **kwds)
-    return function_lib.defun(fn_with_cond)(*canon_args, **canon_kwds)
+    # TODO(jlchu): verify that modification to fn_with_cond works
```

Comment:
please verify before submission, once verified please remove this comment

Reply:
Removed, per discussion that we would verify with GitHub checks passing